packages/babel-helper-create-regexp-features-plugin/src/index.ts

```diff
@@ -71,14 +98,38 @@ export function createRegExpFeaturePlugin({
         const regexpuOptions = generateRegexpuOptions(features);
         if (canSkipRegexpu(node, regexpuOptions)) return;
 
-        const namedCaptureGroups: Record<string, number> = {};
+        const namedCaptureGroups: Record<string, number | number[]> = {
+          __proto__: null,
+        };
+        let hasDuplicateNamedCaptureGroups = false;
         if (regexpuOptions.namedGroups === "transform") {
           regexpuOptions.onNamedGroup = (name, index) => {
-            namedCaptureGroups[name] = index;
+            const prev = namedCaptureGroups[name];
+            if (typeof prev === "number") {
+              hasDuplicateNamedCaptureGroups = true;
+              namedCaptureGroups[name] = [prev, index];
+            } else if (Array.isArray(prev)) {
+              prev.push(index);
+            } else {
+              namedCaptureGroups[name] = index;
+            }
           };
         }
 
-        node.pattern = rewritePattern(node.pattern, node.flags, regexpuOptions);
+        let pattern = rewritePattern(node.pattern, node.flags, regexpuOptions);
+        if (
+          !hasDuplicateNamedCaptureGroups &&
+          hasFeature(features, FEATURES.duplicateNamedCaptureGroups) &&
+          !hasFeature(features, FEATURES.namedCaptureGroups)
+        ) {
+          // If we only transformed named groups because we want to transform duplicates
+          // but there are no duplicates, re-transform the regexp without transforming
+          // named groups.
```

Comment:
Can `regexpu-core` support transforming duplicate named capture groups only? I think it would be useful even out of Babel's scope. And it benefits us, too: we don't have to re-run `regexpu-core`.

Reply:
If transform `(?<a>)|(?<a>)(?<b>)` to `()|()(?<b>)` and then add an `a` property to the `.groups` object in our helper, we get `Object.keys(groups) == ["b", "a"]` instead of `["a", "b"]` :slightly_frowning_face: